providers/src/airflow/providers/google/cloud/openlineage/mixins.py

```diff
@@ -199,115 +201,166 @@ def _get_dataset(self, table: dict, dataset_type: str) -> Dataset:
+    def _get_table_facets_safely(self, table_name: str) -> dict[str, DatasetFacet]:
         try:
-            return self._get_table_schema(table_name)
+            bq_table = self._client.get_table(table_name)
+            return get_facets_from_bq_table(bq_table)
         except Exception as e:
-            self.log.warning("Could not extract output schema from bigquery. %s", e)  # type: ignore[attr-defined]
-        return None
-
-    def _get_table_schema(self, table: str) -> SchemaDatasetFacet | None:
-        bq_table = self._client.get_table(table)
-
-        if not bq_table._properties:
-            return None
-
-        fields = get_from_nullable_chain(bq_table._properties, ["schema", "fields"])
-        if not fields:
-            return None
-
-        return SchemaDatasetFacet(
-            fields=[
-                SchemaDatasetFacetFields(
-                    name=field.get("name"),
-                    type=field.get("type"),
-                    description=field.get("description"),
-                )
-                for field in fields
-            ]
-        )
+            self.log.warning("Could not extract facets from bigquery table: `%s`. %s", table_name, e)  # type: ignore[attr-defined]
+        return {}
 
-    def _get_inputs_and_output_for_query_job(
+    def _get_inputs_and_outputs_for_query_job(
         self, properties: dict
-    ) -> tuple[list[InputDataset], OutputDataset | None]:
+    ) -> tuple[tuple[InputDataset, ...], tuple[OutputDataset, ...]]:
         input_tables = get_from_nullable_chain(properties, ["statistics", "query", "referencedTables"]) or []
         output_table = get_from_nullable_chain(properties, ["configuration", "query", "destinationTable"])
 
-        inputs = [
+        inputs = tuple(
             (self._get_input_dataset(input_table))
             for input_table in input_tables
             if input_table != output_table  # Output table is in `referencedTables` and needs to be removed
-        ]
+        )
 
         if not output_table:
-            return inputs, None
+            return inputs, ()
 
         output = self._get_output_dataset(output_table)
-        if dataset_stat_facet := self._get_statistics_dataset_facet(properties):
+        if dataset_stat_facet := self._get_output_statistics_dataset_facet(properties):
             output.outputFacets = output.outputFacets or {}
             output.outputFacets["outputStatistics"] = dataset_stat_facet
-        if cll_facet := self._get_column_level_lineage_facet(properties, output, inputs):
+        if cll_facet := self._get_column_level_lineage_facet_for_query_job(properties, output, inputs):
             output.facets = output.facets or {}
             output.facets["columnLineage"] = cll_facet
-        return inputs, output
+        return inputs, (output,)
+
+    def _get_inputs_and_outputs_for_load_job(
+        self, properties: dict
+    ) -> tuple[tuple[InputDataset, ...], tuple[OutputDataset, ...]]:
```

Comment:
Maybe tuple of lists? It feels that tuples of tuples have more potential for subtle bugs...